service/matching/taskWriter.go

```diff
@@ -204,7 +201,7 @@ func (w *taskWriter) appendTasks(
 		return resp, nil
 
 	case *persistence.ConditionFailedError:
-		w.tlMgr.Stop()
+		w.tlMgr.signalFatalProblem(w.tlMgr.taskQueueID)
```

Comment:
Probably need some comment here.  What does `signalFatalProblem` means here?  Is it to signal shutdown of TaskQueue?

Reply:
What I've been trying to do over time is to detach the task queue's understanding of an error from which it cannot recover, a "FatalProblem", from the action that will be taken as a result of the same. The taskQueueManager and associated types can raise the signal. It's up to the parent entity (in this case the matching engine) to decide what to do in response.

One thing you might not guess about Stop() is that it removes itself from the parent matching engine. I don't like this "remove yourself" behavior because it puts the control logic in the wrong place (i.e. in the controlled object rather than the controller)

Reply:
BTW, the naming of `shouldUnload` is probably wrong (my own fault). It would be more consistent as `isFatal` in light of the argument above.

Reply:
makes sense.